circus-train-aws/src/test/java/com/hotels/bdp/circustrain/aws/AssumeRoleCredentialProviderTest.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.aws;
+
+import com.amazonaws.auth.AWSCredentials;
+import org.apache.commons.lang.NullArgumentException;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Test;
+
+public class AssumeRoleCredentialProviderTest {
+
+    @Test(expected = NullPointerException.class)
+    public void throwsNullPointerException() {
+        AssumeRoleCredentialProvider provider = new AssumeRoleCredentialProvider(null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void throwsIllegalArgumentException() {
+        AssumeRoleCredentialProvider provider = new AssumeRoleCredentialProvider(new Configuration());
+    }
```

Comment:
Can we add a test for the "happy day" scenario when the conf is available and the property is set?

Reply:
Had a quick look so see if we can mock the stsClient but that's not so easy would go down a path of very specific aws object needed to be mocked. If I don't mock we get an error because of course it can't connect. I think we're gonna have to leave that one like it is

Reply:
OK, yeah, I guess we could come up with some way of passing the client in or having a factory which would by default return the stsClient but for tests we could pass a different one in etc. The code here is very AWS specific and there isn't too much too it so I'm OK with keeping it like this for now.

Reply:
yeah so that's what I did passing in a mocked Client but then to get then to work I had to mock such specific aws behaviour test would be very brittle